src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -279,41 +277,127 @@ private Void showTrackingNotification() {
+
+            List<String> uniquePathParts = getUniquePathParts();
+            for (int i = 0; i < getBasePanelCount(); i++) {
+                BasePanel panel = getBasePanelAt(i);
+                BibDatabaseContext bibDatabaseContext = panel.getBibDatabaseContext();
+                DatabaseLocation databaseLocation = bibDatabaseContext.getLocation();
+                Optional<Path> file = bibDatabaseContext.getDatabasePath();
+                boolean isAutosaveEnabled = Globals.prefs.getBoolean(JabRefPreferences.LOCAL_AUTO_SAVE);
+
+                StringBuilder tabTitle = new StringBuilder();
+                StringBuilder toolTipText = new StringBuilder();
+                StringBuilder windowTitle = new StringBuilder();
+
+                if (file.isPresent()) {
+                    Path databasePath = file.get();
+                    String fileName = databasePath.getFileName().toString();
+                    tabTitle.append(fileName);
+                    windowTitle.append(fileName);
+                    toolTipText.append(databasePath.toAbsolutePath().toString());
+                    if (panel.isModified() && !isAutosaveEnabled) {
+                        tabTitle.append('*');
+                        windowTitle.append('*');
+                    }
+
+                    if (databaseLocation == DatabaseLocation.SHARED) {
+                        tabTitle.append(" \u2013 ");
+                        addSharedDbInformation(tabTitle, bibDatabaseContext);
+                        toolTipText.append(' ');
+                        addSharedDbInformation(toolTipText, bibDatabaseContext);
+                        windowTitle.append(" \u2013 ");
+                        addSharedDbInformation(windowTitle, bibDatabaseContext);
+                    }
+
+                    addModeInfo(toolTipText, bibDatabaseContext);
+                    addModeInfo(windowTitle, bibDatabaseContext);
+
+                    if (panel.isModified() && !isAutosaveEnabled) {
+                        addChangedInformation(toolTipText, fileName);
+                    }
 
-            setWindowTitle();
-            updateAllTabTitles();
+                    String uniquePath = uniquePathParts.get(i);
+                    if (!uniquePath.equals(fileName) && uniquePath.contains(File.separator)) {
+                        // remove filename
+                        uniquePath = uniquePath.substring(0, uniquePath.lastIndexOf(File.separator));
+                        tabTitle.append(" \u2013 ").append(uniquePath);
+                        windowTitle.append(" \u2013 ").append(uniquePath);
+                    }
+                } else {
+                    if (databaseLocation == DatabaseLocation.LOCAL) {
+                        tabTitle.append(Localization.lang("untitled"));
+                        windowTitle.append(Localization.lang("untitled"));
+                        if (bibDatabaseContext.getDatabase().hasEntries()) {
```

Comment:
I would always mark it with a star as JabRef will ask what to do with the new database when it's closed.